Snowflake.Data/Client/SnowflakeDbConnectionPool.cs

```diff
@@ -68,14 +68,25 @@ internal static SnowflakeDbConnection getConnection(string connStr)
             }
             lock (_connectionPoolLock)
             {
-                cleanExpiredConnections(); // clean up all expired connections first.
                 for (int i = 0; i < connectionPool.Count; i++)
                 {
                     if (connectionPool[i].ConnectionString.Equals(connStr))
                     {
                         SnowflakeDbConnection conn = connectionPool[i];
                         connectionPool.RemoveAt(i);
-                        return conn;
+                        long timeNow = DateTimeOffset.Now.ToUnixTimeSeconds();
+                        if (conn._poolTimeout <= timeNow)
+                        {
+                            if (conn.SfSession != null)
+                            {
+                                conn.SfSession.close();
+                            }
+                            return null;
```

Comment:
Do we have only one connection in pool for matching connection string? If not, we need logic to continue and reduce i counter as we remove one element. Think about it.